cmd/common.go

```diff
@@ -73,17 +93,32 @@ func workspaceExists(ctx context.Context, config utils.KongClientConfig, workspa
 
 func getWorkspaceName(workspaceFlag string, targetContent *file.Content) string {
 	if workspaceFlag != targetContent.Workspace && workspaceFlag != "" {
-		cprint.DeletePrintf("Warning: Workspace '%v' specified via --workspace flag is "+
-			"different from workspace '%v' found in state file(s).\n", workspaceFlag, targetContent.Workspace)
+		warning := fmt.Sprintf("Workspace '%v' specified via --workspace flag is "+
+			"different from workspace '%v' found in state file(s).", workspaceFlag, targetContent.Workspace)
+		if isJSONOutput {
+			jsonOutput.Warnings = append(jsonOutput.Warnings, warning)
+		} else {
+			cprint.DeletePrintf("Warning: " + warning + "\n")
+		}
 		return workspaceFlag
 	}
 	return targetContent.Workspace
 }
 
 func syncMain(ctx context.Context, filenames []string, dry bool, parallelism,
-	delay int, workspace string,
+	delay int, workspace string, enableJSONOutput bool,
 ) error {
 	// read target file
+	if enableJSONOutput {
+		isJSONOutput = true
```

Comment:
Since you make use of  the global `isJSONOutput` variable only within this file, I think it'd be better to not create a global variable and instead pass the `enableJSONOutput` value to the functions that need it. This would make it easier to test those functions.

Reply:
done.